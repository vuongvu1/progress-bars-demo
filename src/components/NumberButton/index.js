import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// #region Styles
const Wrapper = styled.button`
  width: 60px;
  background: ${(props) => props.theme.white};
  border-radius: 3px;
  color: ${(props) => (props.isPositive ? props.theme.success : props.theme.error)};
  font-weight: bold;
  margin: 10px;
  padding: 6px 10px;
  cursor: pointer;
  transition: transform 100ms;

  &:hover {
    transform: scale(1.1);
  }
`;
// #endregion Styles

const NumberButton = ({ value, updateValue }) => {
  const isPositive = value > 0;
  const handleOnClick = () => updateValue(value);

  return (
    <Wrapper onClick={handleOnClick} isPositive={isPositive}>
      {isPositive ? `+${value}` : value}
    </Wrapper>
  );
};

NumberButton.propTypes = {
  value: PropTypes.number.isRequired,
  updateValue: PropTypes.func.isRequired,
};

export default NumberButton;
