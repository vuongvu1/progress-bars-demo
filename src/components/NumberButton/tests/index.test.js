import React from 'react';
import { render } from '@testing-library/react';

import NumberButton from '../index';

const renderComponent = (props = {}) => render(<NumberButton {...props} />);

describe('<NumberButton />', () => {
  it('should render <NumberButton />', () => {
    const { container } = renderComponent({
      value: 30,
      updateValue: () => {},
    });
    expect(container).toMatchSnapshot();
  });
});
