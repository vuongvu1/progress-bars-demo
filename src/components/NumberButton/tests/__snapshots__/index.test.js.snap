// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberButton /> should render <NumberButton /> 1`] = `
<div>
  <button
    class="sc-bdVaJa jywRWz"
  >
    +30
  </button>
</div>
`;
